services/head_tracker.go

```diff
@@ -128,7 +128,11 @@ func (ht *HeadTracker) Head() *models.IndexableBlockNumber {
 // Attach registers an object that will have HeadTrackable events fired on occurence,
 // such as Connect.
 func (ht *HeadTracker) Attach(t store.HeadTrackable) string {
-	return ht.attachments.attach(t)
+	rval := ht.attachments.attach(t)
+	if ht.connected.IsSet() {
```

Comment:
Again, now that we connect immediately, things that are subsequently attached after connection, should get a connected event. This should have always been the case.